Makefile

```diff
@@ -92,11 +92,59 @@ validate_all: examples/hpsf* pkg/data/templates/*
 		docker rm 'smoke-refinery'; \
 	fi
 
+.PHONY: .smoke_collector
+#: run smoke test for collector components
+#: Do not use directly, use the smoke target instead
+.smoke_collector:
+	if [ -z "$(FILE)" ]; then \
+		echo "+++ no component file provided, use smoke instead -- exiting"; \
+		exit 1; \
+	fi
+
+	@echo generating collector configs for component $(FILE)
+	mkdir -p tmp
+
+	# generate the configs from the provided file
+	go run ./cmd/hpsf -i ${FILE} -o tmp/collector-config.yaml cConfig || exit 1
+
+	# use yq to remove the usage processor and honeycomb extension from collector config
+	yq -i e \
```

Comment:
Can you confirm my understanding? This removal of usage processor and honeycomb extension is still needed because we don't add these to the collector config until later in the process. Those are not available in this generated config and that's what we're testing, but they are expected in our supervised collector image and so we get failures if we keep those in. However, we are at least confirming that we can run our own image of the collector build with our generated configs.

Reply:
Re-reviewing this and testing with @codeboten , this is correct. The Honeycomb extension uses the opamp extension, so if that is not found, we get an error about a missing opamp extension.
`collector server run finished with error: failed to start extensions: extension "opamp" does not exist`
If we remove honeycomb and try to leave just the usage processor, we get an error because the usage processor is expecting to find the honeycomb extension. So both of these do still need to be removed.
`collector server run finished with error: cannot start pipelines: failed to start "usage" processor: extension "honeycomb" does not exist`

Reply:
Yes and no 😄 

The usage processor requires the honeycomb extension, which requires the opamp extension. Configuring the the opamp extension is complicated because it requires an opamp server (eg beekeeper) to talk to. Removing these components means we don't need the additional setup and can just test the HPSF components without the config tooling chain.